test/tbb/test_concurrent_queue.cpp

```diff
@@ -281,3 +282,211 @@ TEST_CASE("Test clear and dtor with TrackableItem") {
+    srand(static_cast<unsigned>(time(0)));
+    generate(vec_1.begin(), vec_1.end(), rand);
+    generate(vec_2.begin(), vec_2.end(), rand);
+    generate(vec_3.begin(), vec_3.end(), rand);
+    generate(vec_4.begin(), vec_4.end(), rand);
+
+    CQ q1, q2, q3;
+    CQ q4({vec_1, vec_2, vec_3});
+    q3 = {vec_4, vec_2, vec_3};
+
+    q1 = q3;
+    q2 = std::move(q3);
+    CHECK(q3.empty());
+
+    CHECK(q1 != q4);
+    q1.swap(q4);
+    CHECK(q2 == q4);
+
+    swap(q2, q3);
+    CHECK(q2.empty());
+    CHECK(q3 == q4);
+}
+
+//! basic test for copy, move and swap
+//! \brief \ref error_guessing
+TEST_CASE("Test iterator queue") {
+    test_queue_helper<tbb::concurrent_queue<std::vector<int>>, std::vector<int>>();
+    test_queue_helper<tbb::concurrent_bounded_queue<std::vector<int>>, std::vector<int>>();
+}
+
+template <typename queue_type, typename allocator_type>
+void TestMoveQueue() {
+    allocator_type::set_limits(300);
+    queue_type q1, q2;
+    move_support_tests::Foo obj;
+    size_t n1(15), n2(7);
+
+    allocator_type::init_counters();
+    for(size_t i =0; i < n1; i++)
+        q1.push(obj);
+    size_t q1_items_constructed = allocator_type::items_constructed;
+    size_t q1_items_allocated =  allocator_type::items_allocated;
+
+    allocator_type::init_counters();
+    for(size_t i =0; i < n2; i++)
+        q2.push(obj);
+    size_t q2_items_allocated =  allocator_type::items_allocated;
+
+    allocator_type::init_counters();
+    q1 = std::move(q2);
+
+    CHECK(q1_items_allocated == allocator_type::items_freed);
+    CHECK(q1_items_constructed == allocator_type::items_destroyed);
+    CHECK(q2_items_allocated >= allocator_type::items_allocated);
+}
+
+//! move assignment test for equal allocator
+//! \brief \ref error_guessing
+TEST_CASE("Test move queue") {
+    using allocator_type = StaticSharedCountingAllocator<std::allocator<move_support_tests::Foo>>;
+    TestMoveQueue<tbb::concurrent_queue<move_support_tests::Foo, allocator_type>, allocator_type>();
+    TestMoveQueue<tbb::concurrent_bounded_queue<move_support_tests::Foo, allocator_type>, allocator_type>();
+}
+
+template<class T>
+struct stateful_allocator {
+    typedef T value_type;
+    stateful_allocator () = default;
+    int state = 0;
+    template<class U>
+    constexpr stateful_allocator (const stateful_allocator <U>&) noexcept {}
+
+    T* allocate(std::size_t n) {
+        return static_cast<T*>(std::malloc(n * sizeof(T)));
+    }
+
+    void deallocate(T* p, std::size_t n) noexcept {
+        static_cast<void>(n);
+        std::free(p);
+    }
+};
+
+template<class T, class U>
+bool operator==(const stateful_allocator <T>& lhs, const stateful_allocator <U>& rhs) { return &lhs.state == &rhs.state; }
+
+template<class T, class U>
+bool operator!=(const stateful_allocator <T>& lhs, const stateful_allocator <U>& rhs) { return &lhs.state != &rhs.state; }
+
+template <typename queue_type, typename allocator_type>
```

Comment:
Same as above: replace `queue_type` -> `QueueType`, drop `allocator_type` and replace it with `typename QueueType::allocator_type`

Reply:
removed